frontend/src/components/header/MenuOverlay.js

```diff
@@ -96,18 +97,23 @@ class MenuOverlay extends Component {
     if (targetValue) {
       this.setState({
         query: e.target.value,
+        pendingQuery: e.target.value ? true : false,
```

Comment:
why not using `targetValue` instead of `e.target.value`.

Also, i would rename `targetValue` (what a name?!) to something like `query`.